Does not compile on alpine linux

## Brief description of the problem
Thank you for reading this. If I do something wrong, feel free to simply close this issue.

readxl does not compile on Alpine Linux. The standard version from CRAN stops with problems regarding `uid_t` not having been declared. I also checked the current master from github, producing the same error, so I won't this one include here, since it's more complicated to reproduce then, because devtools form CRAN don't work on alpine yet.

---
## Reproduce with:

```bash
Last login: Thu Mar 21 17:23:15 on ttys004
SHIP-NOTE034:~ struckmanns$ docker run -ti artemklevtsov/r-alpine:latest /bin/bash
bash-4.4# R
```
```r
R version 3.5.0 (2018-04-23) -- "Joy in Playing"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages("readxl")
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
also installing the dependencies ‘assertthat’, ‘utf8’, ‘magrittr’, ‘rematch’, ‘cli’, ‘crayon’, ‘fansi’, ‘pillar’, ‘pkgconfig’, ‘rlang’, ‘hms’, ‘prettyunits’, ‘R6’, ‘cellranger’, ‘Rcpp’, ‘tibble’, ‘progress’

trying URL 'https://cloud.r-project.org/src/contrib/assertthat_0.2.1.tar.gz'
Content type 'application/x-gzip' length 12742 bytes (12 KB)
==================================================
downloaded 12 KB

trying URL 'https://cloud.r-project.org/src/contrib/utf8_1.1.4.tar.gz'
Content type 'application/x-gzip' length 218882 bytes (213 KB)
==================================================
downloaded 213 KB

trying URL 'https://cloud.r-project.org/src/contrib/magrittr_1.5.tar.gz'
Content type 'application/x-gzip' length 200504 bytes (195 KB)
==================================================
downloaded 195 KB

trying URL 'https://cloud.r-project.org/src/contrib/rematch_1.0.1.tar.gz'
Content type 'application/x-gzip' length 3719 bytes
==================================================
downloaded 3719 bytes

trying URL 'https://cloud.r-project.org/src/contrib/cli_1.1.0.tar.gz'
Content type 'application/x-gzip' length 40232 bytes (39 KB)
==================================================
downloaded 39 KB

trying URL 'https://cloud.r-project.org/src/contrib/crayon_1.3.4.tar.gz'
Content type 'application/x-gzip' length 658694 bytes (643 KB)
==================================================
downloaded 643 KB

trying URL 'https://cloud.r-project.org/src/contrib/fansi_0.4.0.tar.gz'
Content type 'application/x-gzip' length 266123 bytes (259 KB)
==================================================
downloaded 259 KB

trying URL 'https://cloud.r-project.org/src/contrib/pillar_1.3.1.tar.gz'
Content type 'application/x-gzip' length 103972 bytes (101 KB)
==================================================
downloaded 101 KB

trying URL 'https://cloud.r-project.org/src/contrib/pkgconfig_2.0.2.tar.gz'
Content type 'application/x-gzip' length 6024 bytes
==================================================
downloaded 6024 bytes

trying URL 'https://cloud.r-project.org/src/contrib/rlang_0.3.2.tar.gz'
Content type 'application/x-gzip' length 858244 bytes (838 KB)
==================================================
downloaded 838 KB

trying URL 'https://cloud.r-project.org/src/contrib/hms_0.4.2.tar.gz'
Content type 'application/x-gzip' length 12773 bytes (12 KB)
==================================================
downloaded 12 KB

trying URL 'https://cloud.r-project.org/src/contrib/prettyunits_1.0.2.tar.gz'
Content type 'application/x-gzip' length 8262 bytes
==================================================
downloaded 8262 bytes

trying URL 'https://cloud.r-project.org/src/contrib/R6_2.4.0.tar.gz'
Content type 'application/x-gzip' length 31545 bytes (30 KB)
==================================================
downloaded 30 KB

trying URL 'https://cloud.r-project.org/src/contrib/cellranger_1.1.0.tar.gz'
Content type 'application/x-gzip' length 63857 bytes (62 KB)
==================================================
downloaded 62 KB

trying URL 'https://cloud.r-project.org/src/contrib/Rcpp_1.0.1.tar.gz'
Content type 'application/x-gzip' length 3661123 bytes (3.5 MB)
==================================================
downloaded 3.5 MB

trying URL 'https://cloud.r-project.org/src/contrib/tibble_2.1.1.tar.gz'
Content type 'application/x-gzip' length 311836 bytes (304 KB)
==================================================
downloaded 304 KB

trying URL 'https://cloud.r-project.org/src/contrib/progress_1.2.0.tar.gz'
Content type 'application/x-gzip' length 27369 bytes (26 KB)
==================================================
downloaded 26 KB

trying URL 'https://cloud.r-project.org/src/contrib/readxl_1.3.1.tar.gz'
Content type 'application/x-gzip' length 2056165 bytes (2.0 MB)
==================================================
downloaded 2.0 MB

* installing *source* package ‘assertthat’ ...
** package ‘assertthat’ successfully unpacked and MD5 sums checked
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (assertthat)
* installing *source* package ‘utf8’ ...
** package ‘utf8’ successfully unpacked and MD5 sums checked
** libs
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c as_utf8.c -o as_utf8.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c bytes.c -o bytes.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c context.c -o context.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c init.c -o init.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c render.c -o render.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c render_table.c -o render_table.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c string.c -o string.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c text.c -o text.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8_encode.c -o utf8_encode.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8_format.c -o utf8_format.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8_normalize.c -o utf8_normalize.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8_valid.c -o utf8_valid.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8_width.c -o utf8_width.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c util.c -o util.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/array.c -o utf8lite/src/array.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/char.c -o utf8lite/src/char.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/encode.c -o utf8lite/src/encode.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/error.c -o utf8lite/src/error.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/escape.c -o utf8lite/src/escape.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/graph.c -o utf8lite/src/graph.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/graphscan.c -o utf8lite/src/graphscan.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/normalize.c -o utf8lite/src/normalize.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/render.c -o utf8lite/src/render.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/text.c -o utf8lite/src/text.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/textassign.c -o utf8lite/src/textassign.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/textiter.c -o utf8lite/src/textiter.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -Iutf8lite/src -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8lite/src/textmap.c -o utf8lite/src/textmap.o
ar rcs libcutf8lite.a utf8lite/src/array.o utf8lite/src/char.o utf8lite/src/encode.o utf8lite/src/error.o utf8lite/src/escape.o utf8lite/src/graph.o utf8lite/src/graphscan.o utf8lite/src/normalize.o utf8lite/src/render.o utf8lite/src/text.o utf8lite/src/textassign.o utf8lite/src/textiter.o utf8lite/src/textmap.o
gcc -shared -L/usr/lib/R/lib -L/usr/local/lib -o utf8.so as_utf8.o bytes.o context.o init.o render.o render_table.o string.o text.o utf8_encode.o utf8_format.o utf8_normalize.o utf8_valid.o utf8_width.o util.o -L. -lcutf8lite -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/utf8/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (utf8)
* installing *source* package ‘magrittr’ ...
** package ‘magrittr’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (magrittr)
* installing *source* package ‘rematch’ ...
** package ‘rematch’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (rematch)
* installing *source* package ‘crayon’ ...
** package ‘crayon’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (crayon)
* installing *source* package ‘fansi’ ...
** package ‘fansi’ successfully unpacked and MD5 sums checked
** libs
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c assumptions.c -o assumptions.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c has.c -o has.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c init.c -o init.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c nchar.c -o nchar.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c read.c -o read.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c rnchar.c -o rnchar.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c state.c -o state.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c strip.c -o strip.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c strsplit.c -o strsplit.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c tabs.c -o tabs.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c tohtml.c -o tohtml.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c unhandled.c -o unhandled.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c unique.c -o unique.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utf8.c -o utf8.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c utils.c -o utils.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c wrap.c -o wrap.o
gcc -shared -L/usr/lib/R/lib -L/usr/local/lib -o fansi.so assumptions.o has.o init.o nchar.o read.o rnchar.o state.o strip.o strsplit.o tabs.o tohtml.o unhandled.o unique.o utf8.o utils.o wrap.o -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/fansi/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (fansi)
* installing *source* package ‘pkgconfig’ ...
** package ‘pkgconfig’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (pkgconfig)
* installing *source* package ‘rlang’ ...
** package ‘rlang’ successfully unpacked and MD5 sums checked
** libs
gcc -I"/usr/include/R" -DNDEBUG -I./lib/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c capture.c -o capture.o
gcc -I"/usr/include/R" -DNDEBUG -I./lib/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c export.c -o export.o
gcc -I"/usr/include/R" -DNDEBUG -I./lib/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c internal.c -o internal.o
gcc -I"/usr/include/R" -DNDEBUG -I./lib/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c lib.c -o lib.o
gcc -shared -L/usr/lib/R/lib -L/usr/local/lib -o rlang.so capture.o export.o internal.o lib.o -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/rlang/libs
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** testing if installed package can be loaded
* DONE (rlang)
* installing *source* package ‘R6’ ...
** package ‘R6’ successfully unpacked and MD5 sums checked
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (R6)
* installing *source* package ‘Rcpp’ ...
** package ‘Rcpp’ successfully unpacked and MD5 sums checked
** libs
g++  -I"/usr/include/R" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c Date.cpp -o Date.o
g++  -I"/usr/include/R" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c Module.cpp -o Module.o
g++  -I"/usr/include/R" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c Rcpp_init.cpp -o Rcpp_init.o
g++  -I"/usr/include/R" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c api.cpp -o api.o
g++  -I"/usr/include/R" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c attributes.cpp -o attributes.o
g++  -I"/usr/include/R" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c barrier.cpp -o barrier.o
g++ -shared -L/usr/lib/R/lib -L/usr/local/lib -o Rcpp.so Date.o Module.o Rcpp_init.o api.o attributes.o barrier.o -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/Rcpp/libs
** R
** inst
** byte-compile and prepare package for lazy loading
code for methods in class “C++OverloadedMethods” was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
code for methods in class “RcppClass” was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
code for methods in class “RcppClass” was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (Rcpp)
* installing *source* package ‘cli’ ...
** package ‘cli’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (cli)
* installing *source* package ‘hms’ ...
** package ‘hms’ successfully unpacked and MD5 sums checked
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (hms)
* installing *source* package ‘prettyunits’ ...
** package ‘prettyunits’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (prettyunits)
* installing *source* package ‘pillar’ ...
** package ‘pillar’ successfully unpacked and MD5 sums checked
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** testing if installed package can be loaded
* DONE (pillar)
* installing *source* package ‘progress’ ...
** package ‘progress’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (progress)
* installing *source* package ‘tibble’ ...
** package ‘tibble’ successfully unpacked and MD5 sums checked
** libs
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c coerce.c -o coerce.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c init.c -o init.o
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_DEFAULT_SOURCE -D__USE_MISC  -c matrixToDataFrame.c -o matrixToDataFrame.o
gcc -shared -L/usr/lib/R/lib -L/usr/local/lib -o tibble.so coerce.o init.o matrixToDataFrame.o -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/tibble/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (tibble)
* installing *source* package ‘cellranger’ ...
** package ‘cellranger’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (cellranger)
* installing *source* package ‘readxl’ ...
** package ‘readxl’ successfully unpacked and MD5 sums checked
** libs
g++  -I"/usr/include/R" -DNDEBUG -Iunix -I. -DRCPP_DEFAULT_INCLUDE_CALL=false -I"/usr/local/lib/R/site-library/progress/include" -I"/usr/local/lib/R/site-library/Rcpp/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c RcppExports.cpp -o RcppExports.o
g++  -I"/usr/include/R" -DNDEBUG -Iunix -I. -DRCPP_DEFAULT_INCLUDE_CALL=false -I"/usr/local/lib/R/site-library/progress/include" -I"/usr/local/lib/R/site-library/Rcpp/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c XlsWorkBook.cpp -o XlsWorkBook.o
g++  -I"/usr/include/R" -DNDEBUG -Iunix -I. -DRCPP_DEFAULT_INCLUDE_CALL=false -I"/usr/local/lib/R/site-library/progress/include" -I"/usr/local/lib/R/site-library/Rcpp/include" -I/usr/local/include   -fpic  -g -O2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -D__MUSL__  -c XlsWorkSheet.cpp -o XlsWorkSheet.o
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:51:25: error: 'uid_t' has not been declared
 int chown(const char *, uid_t, gid_t);
                         ^~~~~
/usr/include/unistd.h:51:32: error: 'gid_t' has not been declared
 int chown(const char *, uid_t, gid_t);
                                ^~~~~
/usr/include/unistd.h:52:17: error: 'uid_t' has not been declared
 int fchown(int, uid_t, gid_t);
                 ^~~~~
/usr/include/unistd.h:52:24: error: 'gid_t' has not been declared
 int fchown(int, uid_t, gid_t);
                        ^~~~~
/usr/include/unistd.h:53:26: error: 'uid_t' has not been declared
 int lchown(const char *, uid_t, gid_t);
                          ^~~~~
/usr/include/unistd.h:53:33: error: 'gid_t' has not been declared
 int lchown(const char *, uid_t, gid_t);
                                 ^~~~~
/usr/include/unistd.h:54:33: error: 'uid_t' has not been declared
 int fchownat(int, const char *, uid_t, gid_t, int);
                                 ^~~~~
/usr/include/unistd.h:54:40: error: 'gid_t' has not been declared
 int fchownat(int, const char *, uid_t, gid_t, int);
                                        ^~~~~
/usr/include/unistd.h:107:1: error: 'uid_t' does not name a type
 uid_t getuid(void);
 ^~~~~
/usr/include/unistd.h:108:1: error: 'uid_t' does not name a type
 uid_t geteuid(void);
 ^~~~~
/usr/include/unistd.h:109:1: error: 'gid_t' does not name a type
 gid_t getgid(void);
 ^~~~~
/usr/include/unistd.h:110:1: error: 'gid_t' does not name a type
 gid_t getegid(void);
 ^~~~~
/usr/include/unistd.h:111:20: error: 'gid_t' has not been declared
 int getgroups(int, gid_t []);
                    ^~~~~
/usr/include/unistd.h:112:12: error: 'uid_t' was not declared in this scope
 int setuid(uid_t);
            ^~~~~
/usr/include/unistd.h:112:12: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:113:13: error: 'uid_t' was not declared in this scope
 int seteuid(uid_t);
             ^~~~~
/usr/include/unistd.h:113:13: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:114:12: error: 'gid_t' was not declared in this scope
 int setgid(gid_t);
            ^~~~~
/usr/include/unistd.h:114:12: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:115:13: error: 'gid_t' was not declared in this scope
 int setegid(gid_t);
             ^~~~~
/usr/include/unistd.h:115:13: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:137:14: error: 'uid_t' was not declared in this scope
 int setreuid(uid_t, uid_t);
              ^~~~~
/usr/include/unistd.h:137:14: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:137:21: error: 'uid_t' was not declared in this scope
 int setreuid(uid_t, uid_t);
                     ^~~~~
/usr/include/unistd.h:137:21: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:137:26: error: expression list treated as compound expression in initializer [-fpermissive]
 int setreuid(uid_t, uid_t);
                          ^
/usr/include/unistd.h:138:14: error: 'gid_t' was not declared in this scope
 int setregid(gid_t, gid_t);
              ^~~~~
/usr/include/unistd.h:138:14: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:138:21: error: 'gid_t' was not declared in this scope
 int setregid(gid_t, gid_t);
                     ^~~~~
/usr/include/unistd.h:138:21: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:138:26: error: expression list treated as compound expression in initializer [-fpermissive]
 int setregid(gid_t, gid_t);
                          ^
/usr/include/unistd.h:169:29: error: 'gid_t' does not name a type
 int setgroups(size_t, const gid_t *);
                             ^~~~~
/usr/include/unistd.h:183:15: error: 'uid_t' was not declared in this scope
 int setresuid(uid_t, uid_t, uid_t);
               ^~~~~
/usr/include/unistd.h:183:15: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:183:22: error: 'uid_t' was not declared in this scope
 int setresuid(uid_t, uid_t, uid_t);
                      ^~~~~
/usr/include/unistd.h:183:22: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:183:29: error: 'uid_t' was not declared in this scope
 int setresuid(uid_t, uid_t, uid_t);
                             ^~~~~
/usr/include/unistd.h:183:29: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:183:34: error: expression list treated as compound expression in initializer [-fpermissive]
 int setresuid(uid_t, uid_t, uid_t);
                                  ^
/usr/include/unistd.h:184:15: error: 'gid_t' was not declared in this scope
 int setresgid(gid_t, gid_t, gid_t);
               ^~~~~
/usr/include/unistd.h:184:15: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:184:22: error: 'gid_t' was not declared in this scope
 int setresgid(gid_t, gid_t, gid_t);
                      ^~~~~
/usr/include/unistd.h:184:22: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:184:29: error: 'gid_t' was not declared in this scope
 int setresgid(gid_t, gid_t, gid_t);
                             ^~~~~
/usr/include/unistd.h:184:29: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:184:34: error: expression list treated as compound expression in initializer [-fpermissive]
 int setresgid(gid_t, gid_t, gid_t);
                                  ^
/usr/include/unistd.h:185:15: error: 'uid_t' was not declared in this scope
 int getresuid(uid_t *, uid_t *, uid_t *);
               ^~~~~
/usr/include/unistd.h:185:15: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:185:22: error: expected primary-expression before ',' token
 int getresuid(uid_t *, uid_t *, uid_t *);
                      ^
/usr/include/unistd.h:185:24: error: 'uid_t' was not declared in this scope
 int getresuid(uid_t *, uid_t *, uid_t *);
                        ^~~~~
/usr/include/unistd.h:185:24: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:185:31: error: expected primary-expression before ',' token
 int getresuid(uid_t *, uid_t *, uid_t *);
                               ^
/usr/include/unistd.h:185:33: error: 'uid_t' was not declared in this scope
 int getresuid(uid_t *, uid_t *, uid_t *);
                                 ^~~~~
/usr/include/unistd.h:185:33: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:286:18: note:   'xls::uid_t'
 typedef unsigned uid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:185:40: error: expected primary-expression before ')' token
 int getresuid(uid_t *, uid_t *, uid_t *);
                                        ^
/usr/include/unistd.h:185:40: error: expression list treated as compound expression in initializer [-fpermissive]
/usr/include/unistd.h:186:15: error: 'gid_t' was not declared in this scope
 int getresgid(gid_t *, gid_t *, gid_t *);
               ^~~~~
/usr/include/unistd.h:186:15: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:186:22: error: expected primary-expression before ',' token
 int getresgid(gid_t *, gid_t *, gid_t *);
                      ^
/usr/include/unistd.h:186:24: error: 'gid_t' was not declared in this scope
 int getresgid(gid_t *, gid_t *, gid_t *);
                        ^~~~~
/usr/include/unistd.h:186:24: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:186:31: error: expected primary-expression before ',' token
 int getresgid(gid_t *, gid_t *, gid_t *);
                               ^
/usr/include/unistd.h:186:33: error: 'gid_t' was not declared in this scope
 int getresgid(gid_t *, gid_t *, gid_t *);
                                 ^~~~~
/usr/include/unistd.h:186:33: note: suggested alternative:
In file included from /usr/include/sys/types.h:57:0,
                 from ./libxls/xlstypes.h:40,
                 from ./libxls/xls.h:42,
                 from ColSpec.h:5,
                 from XlsWorkSheet.cpp:2:
/usr/include/bits/alltypes.h:291:18: note:   'xls::gid_t'
 typedef unsigned gid_t;
                  ^~~~~
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:6:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/unistd.h:186:40: error: expected primary-expression before ')' token
 int getresgid(gid_t *, gid_t *, gid_t *);
                                        ^
/usr/include/unistd.h:186:40: error: expression list treated as compound expression in initializer [-fpermissive]
In file included from /usr/local/lib/R/site-library/progress/include/RProgress.h:7:0,
                 from Spinner.h:4,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/include/sys/time.h:18:17: error: field 'it_interval' has incomplete type 'timeval'
  struct timeval it_interval;
                 ^~~~~~~~~~~
/usr/include/sys/time.h:11:26: note: forward declaration of 'struct timeval'
 int gettimeofday (struct timeval *__restrict, void *__restrict);
                          ^~~~~~~
/usr/include/sys/time.h:19:17: error: field 'it_value' has incomplete type 'timeval'
  struct timeval it_value;
                 ^~~~~~~~
/usr/include/sys/time.h:11:26: note: forward declaration of 'struct timeval'
 int gettimeofday (struct timeval *__restrict, void *__restrict);
                          ^~~~~~~
In file included from Spinner.h:4:0,
                 from XlsWorkSheet.h:7,
                 from XlsWorkSheet.cpp:3:
/usr/local/lib/R/site-library/progress/include/RProgress.h: In static member function 'static double RProgress::RProgress::time_now()':
/usr/local/lib/R/site-library/progress/include/RProgress.h:311:20: error: aggregate 'timeval now' has incomplete type and cannot be defined
     struct timeval now;
                    ^~~
make: *** [/usr/lib/R/etc/Makeconf:166: XlsWorkSheet.o] Error 1
ERROR: compilation failed for package ‘readxl’
* removing ‘/usr/local/lib/R/site-library/readxl’

The downloaded source packages are in
	‘/tmp/RtmpJiBbCe/downloaded_packages’
Warning message:
In install.packages("readxl") :
  installation of package ‘readxl’ had non-zero exit status
```

--- a/src/XlsWorkSheet.cpp
+++ b/src/XlsWorkSheet.cpp
@@ -1,3 +1,5 @@
+#include <unistd.h>
+#include <sys/time.h>
 #include <Rcpp.h>
 #include "ColSpec.h"
 #include "XlsWorkSheet.h"

--- a/src/XlsxWorkSheet.cpp
+++ b/src/XlsxWorkSheet.cpp
@@ -1,3 +1,5 @@
+#include <unistd.h>
+#include <sys/time.h>
 #include <Rcpp.h>
 #include "ColSpec.h"
 #include "XlsxWorkSheet.h"

